python/plugins/processing/algs/gdal/GdalAlgorithmProvider.py

```diff
@@ -146,6 +148,7 @@ def loadAlgorithms(self):
             ClipRasterByExtent(),
             ClipRasterByMask(),
             ColorRelief(),
+            CreateCog(),
```

Comment:
To answer your main question.
Why not doing something like this, so it'd be available wherever possible?

```
if int(osgeo.gdal.VersionInfo()) >= 3010000:
    self.algs+= [CreateCog()]
```

Reply:
or `if osgeo.gdal.GetDriverByName('COG') is not None:`